pkg/services/ngalert/api/api_testing.go

```diff
@@ -64,8 +62,8 @@ func (srv TestingApiSrv) RouteTestGrafanaRuleConfig(c *contextmodel.ReqContext,
 		return ErrResp(http.StatusBadRequest, err, "")
 	}
 
-	if !srv.authz.AuthorizeAccessToRuleGroup(c.Req.Context(), c.SignedInUser, ngmodels.RulesGroup{rule}) {
-		return errorToResponse(fmt.Errorf("%w to query one or many data sources used by the rule", accesscontrol.ErrAuthorization))
+	if err := srv.authz.AuthorizeAccessToRuleGroup(c.Req.Context(), c.SignedInUser, ngmodels.RulesGroup{rule}); err != nil {
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to authorize access to rule group", err)
```

Comment:
oh, I think it is better to do the opposite because ErrOrFallback lets us return some meaningful message in the response if error does not happen to be errutils' one. WDYT?